### ISSUE TITLE
Zenoss state changes production state even when test=true

### ISSUE THREAD

**Issue by rossengeorgiev** [saltstack/salt#42947]
### Description of Issue
Using the `zenoss.monitored` and running it with `test=true` will still change the production state.

### Steps to Reproduce Issue
If the device already exist in Zenoss the code will check the state and set it to whatever is specified, without checking `test` parameter

https://github.com/saltstack/salt/blob/af0c8ffe36c695b9e23a13a3bc2645ed528c1da7/salt/states/zenoss.py#L52-L72

**Comment by garethgreenaway** [comment]
@rossengeorgiev Thanks for the report, test=True should definitely be honored.  we'll get this fixed.  Can you provide a version report, `salt --versions-report`.

**Comment by rossengeorgiev** [comment]
@garethgreenaway heres the versions report

```yaml
Salt Version:
           Salt: 2016.11.5

Dependency Versions:
           cffi: 1.6.0
       cherrypy: Not Installed
       dateutil: 1.5
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: 0.24.6
        libnacl: Not Installed
       M2Crypto: 0.21.1
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: Not Installed
   pycryptodome: 3.4.3
         pygit2: 0.24.2
         Python: 2.7.5 (default, Nov  6 2016, 00:28:07)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: centos 7.3.1611 Core
        machine: x86_64
        release: 3.10.0-514.21.2.el7.x86_64
         system: Linux
        version: CentOS Linux 7.3.1611 Core

**Comment by rossengeorgiev** [comment]
Fixed released in `>=2016.11.8` and `>=2017.7.2`
